### PR TITLE
Onboard IaaS server volume commands

### PR DESCRIPTION
**Pull request by Fyusel**
Supported commands: attach, detach, list, describe, update

### CODE REVIEW

File: internal/cmd/beta/server/volume/volume.go [hunk 1]
```diff
@@ -0,0 +1,34 @@
...
+	"github.com/stackitcloud/stackit-cli/internal/pkg/utils"
+
+	"github.com/spf13/cobra"
+)
+
+func NewCmd(p *print.Printer) *cobra.Command {
+	cmd := &cobra.Command{
+		Use:   "volume",
+		Short: "Provides functionality for Server Volume",
+		Long:  "Provides functionality for Server Volume.",
```
Comment (joaopalet) [inline internal/cmd/beta/server/volume/volume.go:20]:
```suggestion
		Short: "Provides functionality for Server volumes",
		Long:  "Provides functionality for Server volumes.",
```

File: internal/cmd/beta/server/volume/describe/describe.go [hunk 2]
```diff
@@ -0,0 +1,150 @@
...
+
+func NewCmd(p *print.Printer) *cobra.Command {
+	cmd := &cobra.Command{
+		Use:   "describe",
+		Short: "Describes a server volume",
+		Long:  "Describes a server volume.",
+		Args:  args.SingleArg(volumeIdArg, utils.ValidateUUID),
+		Example: examples.Build(
+			examples.NewExample(
+				`Describes server volume with volume id "xxx" and server id "yyy"`,
```
Comment (joaopalet) [inline internal/cmd/beta/server/volume/describe/describe.go:41]:
```suggestion
				`Get details of the attachement of volume with ID "xxx" to server with ID "yyy"`,
```

Comment (joaopalet) [reply internal/cmd/beta/server/volume/describe/describe.go:41]: We usually use `Get details ...` for these `describe` examples

File: internal/cmd/beta/server/volume/detach/detach.go [hunk 3]
```diff
@@ -0,0 +1,127 @@
...
+
+func NewCmd(p *print.Printer) *cobra.Command {
+	cmd := &cobra.Command{
+		Use:   "detach",
+		Short: "Detaches a volume from a server",
+		Long:  "Detaches a volume from a server.",
+		Args:  args.SingleArg(volumeIdArg, utils.ValidateUUID),
+		Example: examples.Build(
+			examples.NewExample(
+				`Detaches a volume with id "xxx" from a server with id "yyy"`,
```
Comment (joaopalet) [inline internal/cmd/beta/server/volume/detach/detach.go:40]:
```suggestion
				`Detaches a volume with ID "xxx" from a server with ID "yyy"`,
```

File: internal/cmd/beta/server/volume/list/list.go [hunk 4]
```diff
@@ -0,0 +1,144 @@
...
+		Short: "Lists all server volumes",
+		Long:  "Lists all server volumes.",
+		Args:  args.NoArgs,
+		Example: examples.Build(
+			examples.NewExample(
+				`List all volumes for a server with id "xxx"`,
+				"$ stackit beta server volume list --server-id xxx"),
+			examples.NewExample(
+				`List all volumes for a server with id "xxx" in JSON format`,
+				"$ stackit beta server volumes list --server-id xxx --output-format json"),
```
Comment (joaopalet) [inline internal/cmd/beta/server/volume/list/list.go:44]:
```suggestion
				`List all volumes for a server with ID "xxx"`,
				"$ stackit beta server volume list --server-id xxx"),
			examples.NewExample(
				`List all volumes for a server with ID "xxx" in JSON format`,
				"$ stackit beta server volumes list --server-id xxx --output-format json"),
```

File: internal/cmd/beta/server/volume/update/update.go [hunk 5]
```diff
@@ -0,0 +1,161 @@
...
+
+func NewCmd(p *print.Printer) *cobra.Command {
+	cmd := &cobra.Command{
+		Use:   "update",
+		Short: "Updates an attached volume of a server",
+		Long:  "Updates an attached volume of a server.",
+		Args:  args.SingleArg(volumeIdArg, utils.ValidateUUID),
+		Example: examples.Build(
+			examples.NewExample(
+				`Updates a volume with id "xxx" of a server with id "yyy" and enables delete on termination`,
```
Comment (joaopalet) [inline internal/cmd/beta/server/volume/update/update.go:46]:
```suggestion
				`Updates a volume with ID "xxx" of a server with ID "yyy" and enables delete on termination`,
```

File: internal/cmd/beta/server/volume/describe/describe.go (hunk 2 above)
Comment (joaopalet) [reply internal/cmd/beta/server/volume/describe/describe.go:41]: Same for the other examples below

File: internal/cmd/beta/server/volume/attach/attach.go [hunk 6]
```diff
@@ -0,0 +1,165 @@
...
+		Long:  "Attaches a volume to a server.",
+		Args:  args.SingleArg(volumeIdArg, utils.ValidateUUID),
+		Example: examples.Build(
+			examples.NewExample(
+				`Attaches a volume with id "xxx" to a server with id "yyy"`,
+				`$ stackit beta server volume attach xxx --server-id yyy`,
+			),
+			examples.NewExample(
+				`Attaches a volume with id "xxx" to a server with id "yyy" and delete on termination is enabled`,
+				`$ stackit beta server volume attach xxx --server-id yyy --delete-on-termination`,
```
Comment (joaopalet) [inline internal/cmd/beta/server/volume/attach/attach.go:51]:
```suggestion
				`Attach a volume with ID "xxx" to a server with ID "yyy"`,
				`$ stackit beta server volume attach xxx --server-id yyy`,
			),
			examples.NewExample(
				`Attach a volume with ID "xxx" to a server with ID "yyy" and enable deletion on termination`,
				`$ stackit beta server volume attach xxx --server-id yyy --delete-on-termination`,
```

File: internal/cmd/beta/server/volume/attach/attach.go [hunk 7]
```diff
@@ -0,0 +1,165 @@
...
+			return outputResult(p, model.OutputFormat, volumeLabel, serverLabel, resp)
+		},
+	}
+	configureFlags(cmd)
+	return cmd
+}
+
+func configureFlags(cmd *cobra.Command) {
+	cmd.Flags().Var(flags.UUIDFlag(), serverIdFlag, "Server ID")
+	cmd.Flags().BoolP(deleteOnTerminationFlag, "b", defaultDeleteOnTermination, "Delete the volume during the termination of the server. Defaults to false.")
```
Comment (joaopalet) [inline internal/cmd/beta/server/volume/attach/attach.go:103]: Isn't the default value already shown after the flag description when running the `--help` on the command?

File: internal/cmd/beta/server/volume/describe/describe.go [hunk 8]
```diff
@@ -0,0 +1,150 @@
...
+	*globalflags.GlobalFlagModel
+	ServerId *string
+	VolumeId string
+}
+
+func NewCmd(p *print.Printer) *cobra.Command {
+	cmd := &cobra.Command{
+		Use:   "describe",
+		Short: "Describes a server volume",
+		Long:  "Describes a server volume.",
```
Comment (joaopalet) [inline internal/cmd/beta/server/volume/describe/describe.go:37]:
```suggestion
		Short: "Describes a server volume attachement",
		Long:  "Describes a server volume attachement.",
```

File: internal/cmd/beta/server/volume/attach/attach.go (hunk 7 above)
Comment (Fyusel) [reply internal/cmd/beta/server/volume/attach/attach.go:103]: Hm, currently not but in comparison to the --nic-security flag for the network-interface command it is shown. I will take a look at it :+1:

Comment (Fyusel) [reply internal/cmd/beta/server/volume/attach/attach.go:103]: I see the difference: If the default value is false it is not printed, if the default value is true it is printed (automatically).

Shall we keep the info in the description or skip it?

Comment (joaopalet) [reply internal/cmd/beta/server/volume/attach/attach.go:103]: I think in this case we can skip it

**Review by GokceGK — APPROVED** [review]

**Pull request merged**